SU2_CFD/include/fluid/CFluidScalar.hpp

```diff
@@ -91,6 +91,11 @@ class CFluidScalar final : public CFluidModel {
    */
   su2double ComputeMeanSpecificHeatCp(const su2double* val_scalars);
 
+  /*!
+   * \brief Compute Enthalpy given the temperature and scalars.
+   */
+  su2double ComputeEnthalpyFromT(const su2double val_temperature, const su2double* val_scalars);
```

Comment:
Why doesn't this use TDState_T

Reply:
because we compute the temperature from the enthalpy and the species mass fractions. Here it can be computed directly but, as the final goal is to simulated reacting flows, we need to use newton-raphson for computing the temperature, that will come in the next pull request when we can use Cantera for that purpose.

Reply:
And why don't you compute the enthalpy in that function too instead of adding another function?
The idea with generic classes like CFluidModel is to have a concise abstract interface. Having specific functions that only a few subclasses implement defeats the purpose of a generic interface.